pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
I probably should have made at least some effort to understand the actual meaning of some of the remarks instead of immediately nitpicking the most literal interpretation.

Re: pending PRs: I did argue the context, but in general you are right. I even tried to search for small (in terms of meaningful lines changed) PRs approved by you, assuming there could be a few quick merges for little reviewing effort, but I haved failed to make GitHub usefully process any one of these two conditions — am I just failing GitHub usage check, or am I failing but it is also impossible?

I do try to read mentions even in the review-burnout stage (which is, unfortunately, what happens every time I try to actively do anything with more than 20 PRs in a single day), feel free to use that fact in case some PR looks like a no-brainer-merge to you (of course I might disagree, but average outcome could be still positive)